### PR TITLE
code monitoring: log frontend events

### PR DESCRIPTION
**Pull request by limitedmage**
Log frontend events as described in [RFC 700](https://docs.google.com/document/d/1QpB2tAginNjvprW51A4F7bT8kHU0SqceBIuGXEN22zk/edit#heading=h.yltlv6dgwggx).

| Event description | Logged event name | Notes |
|---|---|---|
| Code monitoring homepage viewed | `CodeMonitoringViewed` | Pre-existing event, not outlined in RFC 700. Previously called `ViewCodeMonitoringPage` |
| Code monitor getting started page viewed | `CodeMonitoringGettingStartedPageViewed` | |
| Code monitor library item clicked | `CodeMonitoringExampleMonitorClicked` | |
| Code monitor creation started | `CreateCodeMonitorPageViewed` | Pre-existing event, previously `ViewCreateCodeMonitorPage` |
| Code monitor created | `CreateCodeMonitorFormSubmitted` | Pre-existing event |
| Code monitor update started | `ManageCodeMonitorPageViewed` | Pre-existing event, previously `ViewManageCodeMonitorPage` |
| Code monitor updated | `ManageCodeMonitorFormSubmitted` | |
| Code monitor deleted | `ManageCodeMonitorDeleteSubmitted` | |
| Logs page viewed | `CodeMonitoringLogsPageViewed` | |

## Test plan

Trigger the events listed and verify they are being sent to the server by analyzing the network tab in the browser's dev tools.

## App preview:

- [Web](https://sg-web-jp-cmlogevents.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-sbuozvhbtz.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by camdencheek** [comment]
If it's helpful, these are the current events we send pings for: 
```
            'ViewCodeMonitoringPage',
            'ViewCreateCodeMonitorPage',
            'ViewManageCodeMonitorPage',
            'CodeMonitorEmailLinkClicked'
```

**Comment by camdencheek** [comment]
> are being changed to ...PageViewed

Works for me. Could you document which event names you changed in this PR so I can query for both the old and the new events? We don't want pings to show a weird drop because we changed the event names.

**Comment by camdencheek** [comment]
@limitedmage is this ready for review?

**Comment by limitedmage** [comment]
> @limitedmage is this ready for review?

It is!

File: client/web/src/enterprise/code-monitoring/CodeMonitoringPage.tsx [hunk 1]
```diff
@@ -87,6 +85,21 @@ export const CodeMonitoringPage: React.FunctionComponent<React.PropsWithChildren
...
+    // Log page view and selected tab
+    useEffect(() => eventLogger.logPageView('CodeMonitoring'), [])
+    useEffect(() => {
+        if (userHasCodeMonitors !== LOADING) {
+            switch (currentTab) {
+                case 'getting-started':
+                    eventLogger.log('CodeMoitoringGettingStartedPageViewed')
+                    break
+                case 'logs':
+                    eventLogger.log('CodeMoitoringLogsPageViewed')
```
Comment (camdencheek) [inline client/web/src/enterprise/code-monitoring/CodeMonitoringPage.tsx:97]:
```suggestion
                    eventLogger.log('CodeMonitoringGettingStartedPageViewed')
                    break
                case 'logs':
                    eventLogger.log('CodeMonitoringLogsPageViewed')
```

Comment (camdencheek) [reply client/web/src/enterprise/code-monitoring/CodeMonitoringPage.tsx:97]: Also, why `logPageView('CodeMonitoring')` above, but not `logPageView('CodeMonitoringLogsPage')` here?

File: client/web/src/enterprise/code-monitoring/CodeMonitoringPage.tsx [hunk 2]
```diff
@@ -87,6 +85,21 @@ export const CodeMonitoringPage: React.FunctionComponent<React.PropsWithChildren
         }
     }, [currentTab, testForceTab])
 
+    // Log page view and selected tab
+    useEffect(() => eventLogger.logPageView('CodeMonitoring'), [])
```
Comment (camdencheek) [inline client/web/src/enterprise/code-monitoring/CodeMonitoringPage.tsx:89]: Shouldn't this be `logPageView('CodeMonitoringPage')`?

**Review by camdencheek — CHANGES_REQUESTED** [review]

**Pull request merged**